variety/Util.py

```diff
@@ -875,10 +881,50 @@ def safe_unlink(filepath):
         except Exception:
             logger.exception(lambda: "Could not delete {}, ignoring".format(filepath))
 
+    @staticmethod
+    def copy_with_replace(from_path, to_path, search_replace_map):
+        with open(from_path, "r") as file:
+            data = file.read()
+        for search, replace in search_replace_map.items():
+            data = data.replace(search, replace)
+        with open(to_path + ".partial", "w") as file:
+            file.write(data)
+            file.flush()
+        os.rename(to_path + ".partial", to_path)
```

Comment:
Instead of doing manual replaces, might Python string templates do the job? It's a more standard format that supports automatically filling in substitute fields: https://docs.python.org/3/library/string.html#template-strings

Reply:
Interesting idea. Though I like that the search and replace method is more generic - it can be used for example to replace `~/.config/variety` inside `variety.conf` and `set_wallpaper` and we use it like this in `prepare_config_folder`. I.e. it does not put restrictions on the format for whatever will be replaced. But it's true that if we replace the files being copied_with_replace with actual templates, we could use string.Template.

However, a downside with this is that the current `set_wallpaper` is actually a valid script, you can run and test it on its own, while otherwise it would be just a template file with bash-illegal Python variables in it and will not be runnable on its own.

Reply:
Maybe we could use environment variables instead of touching the script file itself. I think it's neat how set_wallpaper can function on its own too